sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -17,27 +17,104 @@ using namespace Azure::Core::Http;
 using namespace Azure::Core::Http::Policies;
 using namespace Azure::Core::Http::Policies::_internal;
 
-std::shared_ptr<HttpTransport> Azure::Core::Http::Policies::_detail::GetTransportAdapter()
-{
-  // The order of these checks is important so that WinHTTP is picked over libcurl on Windows, when
-  // both are defined.
+namespace Azure { namespace Core { namespace Http { namespace Policies { namespace _detail {
+  namespace {
+    bool AnyTransportOptionsSpecified(TransportOptions const& transportOptions)
+    {
+      return !(
+          transportOptions.HttpProxy.empty() && transportOptions.ProxyPassword.empty()
+          && transportOptions.ProxyUserName.empty());
+    }
+    //    std::once_flag createTransportOnce;
+    //    std::shared_ptr<HttpTransport> defaultTransport;
+  } // namespace
+
+  std::shared_ptr<HttpTransport> GetTransportAdapter(TransportOptions const& transportOptions)
+  {
+    // The order of these checks is important so that WinHTTP is picked over libcurl on
+    // Windows, when both are defined.
 #if defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
-  return ::AzureSdkGetCustomHttpTransport();
+    return ::AzureSdkGetCustomHttpTransport();
 #elif defined(BUILD_TRANSPORT_WINHTTP_ADAPTER)
-  return std::make_shared<Azure::Core::Http::WinHttpTransport>();
+    // Since C++11: If multiple threads attempt to initialize the same static local variable
+    // concurrently, the initialization occurs exactly once. We depend on this behavior to ensure
+    // that the singleton defaultTransport is correctly initialized.
+    static std::shared_ptr<HttpTransport> defaultTransport(std::make_shared<WinHttpTransport>());
+    if (AnyTransportOptionsSpecified(transportOptions))
+    {
+      WinHttpTransportOptions httpOptions;
+      if (!transportOptions.HttpProxy.empty())
+      {
+        std::string proxyString;
+        proxyString = "http=" + transportOptions.HttpProxy;
+        proxyString += ";";
+        proxyString += "https=" + transportOptions.HttpProxy;
+        httpOptions.ProxyInformation = proxyString;
```

Comment:
It looks weird that we are setting http and https to the same string...  will that work?  
Maybe we should only set https and don't have a http proxy

Reply:
THis is what WinHTTP expects for proxy strings. 

Adding the following text:
```c++
// WinHTTP proxy strings are semicolon separated elements, each of which
// has the following format:
//  ([<scheme>=][<scheme>"://"]<server>[":"<port>])
```